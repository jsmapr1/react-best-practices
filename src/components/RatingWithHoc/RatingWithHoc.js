import React from 'react';

import withOpen from '../withOpen/withOpen';
import '../ActiveRatings/rating.css';

export function Rating({ handleClick, open, rating }) {
  return (
    <div key={rating.id} className="rating">
      <h3>Rating</h3>
      {[...Array(rating.rating)].map((item, index) => (<span key={index}/>))}
      {!open && <div><a onClick={() => handleClick(rating.id)}>See Reviewer</a></div>}
      {open && <div onClick={() => handleClick(rating.id)}>Reviewer: {rating.reviewer}</div>}
    </div>
  )
}

export default withOpen(Rating);
