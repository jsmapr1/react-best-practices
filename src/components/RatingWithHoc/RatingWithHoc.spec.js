import React from 'React';
import { shallow } from 'enzyme';

import { Rating } from './RatingWithHoc';

describe('RatingsWithHoc', () => {
  it('should show not show reviewer if open is false', () => {
    const rating = {
      id: 'foo',
      reviewer: 'Bill',
      stars: 5
    }
    const wrapper = shallow(
      <Rating
        rating={rating}
        handleClick={() => {}}
        open={false}
      />
    )

    expect(wrapper).toMatchSnapshot();
  })

  it('should show show reviewer if open is true', () => {
    const rating = {
      id: 'foo',
      reviewer: 'Bill',
      rating: 5
    }
    const wrapper = shallow(
      <Rating
        rating={rating}
        handleClick={() => {}}
        open
      />
    )

    expect(wrapper).toMatchSnapshot();
  })

})
