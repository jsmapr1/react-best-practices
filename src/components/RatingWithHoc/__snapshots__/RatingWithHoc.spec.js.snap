// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsWithHoc should show not show reviewer if open is false 1`] = `
<div
  className="rating"
  key="foo"
>
  <h3>
    Rating
  </h3>
  <span
    key="0"
  />
  <span
    key="1"
  />
  <span
    key="2"
  />
  <span
    key="3"
  />
  <span
    key="4"
  />
  <div>
    <a
      onClick={[Function]}
    >
      See Reviewer
    </a>
  </div>
</div>
`;

exports[`RatingsWithHoc should show show reviewer if open is true 1`] = `
<div
  className="rating"
  key="foo"
>
  <h3>
    Rating
  </h3>
  <span
    key="0"
  />
  <span
    key="1"
  />
  <span
    key="2"
  />
  <span
    key="3"
  />
  <span
    key="4"
  />
  <div
    onClick={[Function]}
  >
    Reviewer: 
    Bill
  </div>
</div>
`;
